web/js/containers/sidebar/events.js

```diff
@@ -183,8 +207,17 @@ function mapStateToProps(state) {
   let visibleEvents = {};
   const events = lodashGet(requestedEvents, 'response');
   const sources = lodashGet(requestedEventSources, 'response');
+  const eventCategories = (events || []).reduce((categories, event) => {
```

Comment:
Maybe this could be put into `eventsRequestReducer` so this loop is only run once?

Reply:
Looking at this more closely, it felt more like a value I was trying to derive from an existing state value, so I went with a memoized state selector instead, which will still only run once.